src/components/Dashboard/DashboardCard/DashboardCardDropdownItem.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { noop } from 'patternfly-react';
+
+import { Dropdown } from '../../Form/Dropdown';
+
+export class DashboardCardDropdownItem extends React.PureComponent {
+  render() {
+    const { id, dropdownOptions, onDropdownChange, dropdownType, selectedOption, disabled } = this.props;
+
+    return (
+      <Dropdown
+        id={id}
+        value={selectedOption}
+        onChange={newValue => onDropdownChange(dropdownType, newValue)}
```

Comment:
this is unnecessary - this should be handled by the component which uses it. Btw is this component really necessary? - we could move the margin css to the parent component DashboardCardActionsBody which will apply it to its children.

Reply:
@suomiy this is being handled by the component only. We are passing the function definition which should be called when dropdown option changes and just mapping that function in the `Dropdown`

Reply:
You are passing a function and a value which you want the function to be called with.

That is passing a responsibility on something which doesn't have to have that responsibility. You might as well just pass a function/lambda with this binded value. 

This will be also easier to read from TopConsumers.

Reply:
@suomiy Can you please explain with an example here

Reply:
```jsx
      <DashboardCardDropdownItem
        dropdownOptions={sortBy}
        onDropdownChange={this.changeState}
        dropdownType="sortBy"
        selectedOption={this.state.sortBy}
      />
```

vs

```jsx
      <DashboardCardDropdownItem
        dropdownOptions={sortBy}
        onDropdownChange={value => this.changeState("sortBy", value)}
        selectedOption={this.state.sortBy}
      />
```

is more readable and there is less Magic involved

Reply:
I would rather discard that component altogether though as mentioned before

Reply:
@suomiy Will take care of this function binding change in the follow up PR. There are other tasks, which is being blocked by this. Hence, need to merge this as soon as possible.

Reply:
these are pretty small changes, and IMO we could tackle them here 

\+ this is a refactoring so it makes sense to have them in this PR